subsys/bluetooth/nrf_bt_adv/core.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/nrf_bt_adv.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nrf_bt_adv, CONFIG_NRF_BT_ADV_LOG_LEVEL);
+
+enum provider_set {
+	PROVIDER_SET_AD,
+	PROVIDER_SET_SD
+};
+
+static void get_section_ptrs(enum provider_set set,
+			     const struct nrf_bt_adv_data_provider **start,
```

Comment:
probably can add const
const type * const * var...

Reply:
Function assigns `*start` and `*end`, so the second const cannot be introduced